### PR TITLE
wallaby for writing tests

### PR DESCRIPTION
**Pull request by spirosikmd**
Introduced [wallaby](https://wallabyjs.com/) and jasmine to simplify writing unit tests, one new test was written, for the rest refactoring is needed and split in multiple files.

### CODE REVIEW

**Comment by spirosikmd** [comment]
There is an inconsistency between the test environment of karma and jasmine wallaby, so this needs a bit more work.

File: wallaby.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = function(wallaby) {
```
Comment (shuisman) [inline wallaby.js]: should this file be renamed to wallaby.conf.js? following the config filename structure we see at karma?

**Comment by spirosikmd** [comment]
@shuisman @tzengerink `Object.assign` was not the problem. When running tests in travis we need to include the babel polyfill for es6 in karma. So we can use `Object.assign` and any other es6 feature.

File: test/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      host: 'data.usabilla.com',
+      iterator: true
+    });
+  });
+
+  it('has resources instantiated', function() {
+    expect(this.usabilla.websites instanceof resources.WebsitesProduct).toBe(true);
+    expect(this.usabilla.email instanceof resources.EmailProduct).toBe(true);
+    expect(this.usabilla.apps instanceof resources.AppsProduct).toBe(true);
+  });
```
Comment (shuisman) [inline test/index.spec.js]: nice!

File: test/resources.spec.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+describe('Resources', function() {
+  describe('WebsitesProduct', function() {
+    beforeEach(function() {
+      this.websitesProduct = new resources.WebsitesProduct();
+    });
+
+    it('new', function() {
+      expect(true).toEqual(true);
+    });
+  });
```
Comment (shuisman) [inline test/resources.spec.js]: should we remove this? It wa part of the boiler template?!

**Pull request merged**